src/main/java/main/controllers/Project/ResultController.java

```diff
@@ -167,4 +179,28 @@ private void fillResultSteps(TestResultDto result) throws AqualityException {
         stepResultTemplate.setProject_id(result.getProject_id());
         result.setSteps(stepResultController.get(stepResultTemplate));
     }
+
+    public Map<String, Integer> matchIssues(Integer testResultId) throws AqualityException {
+        TestResultDto testResultTemplate = new TestResultDto();
+        testResultTemplate.setId(testResultId);
+        List<TestResultDto> testResults = this.get(testResultTemplate);
+        testResults = testResults.stream().filter(x -> x.getFinal_result_id() != 2 && x.getFail_reason() != null && x.getIssue_id() == null).collect(Collectors.toList());
```

Comment:
add constant for '2' value or reuse existing so it won't look like a magic number or even create a function to get filtered list